src/textual/_styles_cache.py

```diff
@@ -225,8 +225,9 @@ def render(
                 self._cache[y] = strip
             else:
                 strip = self._cache[y]
-            if filter:
-                strip = strip.apply_filter(filter)
+            if filters:
+                for filter in filters:
+                    strip = strip.apply_filter(filter, background)
```

Comment:
How about just trying to loop?
Not sure if your version is an optimisation for the case when there are no filters?

```suggestion
            for filter in filters:
                strip = strip.apply_filter(filter, background)
```

Reply:
filters may be None